spec/controllers/api/customers_controller_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+
+module Api
+  describe CustomersController, type: :controller do
+    include AuthenticationWorkflow
+    render_views
+
+    let(:user) { create(:user) }
+    let(:customer) { create(:customer, user: user) }
+    let(:params) { { format: :json, id: customer.id, customer: { code: '123' } } }
+
+    describe "#update" do
+      context "as a user who is not associated with the customer" do
+        before do
+          allow(controller).to receive(:spree_current_user) { create(:user) }
```

Comment:
Why do you create a new user here instead of using `user` like below? Ah, you want another user to test the permission. I would probably add `let(:other_user)`.